platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileCollectionMergeTest.java

```diff
@@ -383,40 +417,50 @@ void testMergeUpdateSnapshotRestore(final int testParam) throws Throwable {
             }
-            store.endWriting(0, index.size()).setFileAvailableForMerging(true);
+            store.endWriting(0, index.size()).setFileCompleted();
         }
         // Start compaction
-        final Semaphore mergingPaused = new Semaphore(1);
         // Test scenario 0: start merging with mergingPaused semaphore locked, so merging
         // won't proceed more than to the first index update
         if (testParam == 0) {
-            mergingPaused.acquire();
+            store.pauseCompaction();
         }
-        final int filesCountBeforeMerge = store.getAllFilesAvailableForMerge().size();
+        final int filesCountBeforeMerge = store.getAllCompletedFiles().size();
         assertEquals(numFiles, filesCountBeforeMerge);
         final CountDownLatch mergeCompleteLatch = new CountDownLatch(1);
-        final AtomicReference<Throwable> mergeThreadExceptions = new AtomicReference<>();
-        new Thread(() -> {
-                    try {
-                        final List<DataFileReader<long[]>> filesToMerge = store.getAllFilesAvailableForMerge();
-                        assertEquals(numFiles, filesToMerge.size());
-                        store.mergeFiles(index, filesToMerge, mergingPaused);
-                        // 2 = merged file + new file with updated values below, assuming there is
-                        // a single merged file (there may be many, if data is large, but this shouldn't
-                        // be the case in this test
-                        assertEquals(2, store.getAllFilesAvailableForMerge().size());
-                    } catch (final Throwable t) {
-                        mergeThreadExceptions.set(t);
-                    } finally {
-                        mergeCompleteLatch.countDown();
-                    }
-                })
-                .start();
+        final CountDownLatch newFileWriteCompleteLatch = new CountDownLatch(1);
+        final ExecutorService exec = Executors.newSingleThreadExecutor();
+        final Future<?> f = exec.submit(() -> {
+            try {
+                final List<DataFileReader<long[]>> filesToMerge = store.getAllCompletedFiles();
+                assertEquals(numFiles, filesToMerge.size());
+                store.compactFiles(index, filesToMerge);
+                // Wait for the new file to be available. Without this wait, there
+                // may be 1 or 2
+                // files available for merge, as this thread may be complete before
+                // endWriting()
+                // below is called
+                newFileWriteCompleteLatch.await();
+                // 2 = new file with updated values below + one or two files created
+                // during merge,
+                // it depends on where pauseCompaction() happens inside
+                // compactFiles() above
+                assertTrue(List.of(2, 3).contains(store.getAllCompletedFiles().size()));
+            } catch (final Exception e) {
```

Comment:
This probably isn't necessary, the lambda generated code should wrap any exception thrown automatically, and double-wrapping doesn't appear to be needed.

Reply:
If the handler is removed:
```
/ws/hashgraph/hedera-services-2/platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileCollectionMergeTest.java:437: error: unreported exception IOException; must be caught or declared to be thrown
                store.compactFiles(index, filesToMerge);
                                  ^
/ws/hashgraph/hedera-services-2/platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileCollectionMergeTest.java:443: error: unreported exception InterruptedException; must be caught or declared to be thrown
                newFileWriteCompleteLatch.await();
                                               ^
2 errors
```

Reply:
Ahh, thanks.  That's different than I've seen, but most of my use of lambdas has been via the interfaces, rather than anonymous blocks.